### PR TITLE
add text_justification and test text_justification

### PR DESCRIPTION
**Pull request by LJXLJXLJX**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [X] **If creating a new file :**
  - [ ] added links to it in the README files ?
  - [X] included tests with it ?
  - [X] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/strings/text_justification.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            row+=' '*(maxWidth-len(row))
+        ret.append(row)
+        # after a row , reset those value
+        rowLen=0
+        rowWords=[]
+        isFirstWord=True
+    return  ret
+
+
+if __name__=='__main__':
```
Comment (goswami-rahul) [inline algorithms/strings/text_justification.py]: You can remove this execution part from the file. This should go on the doc-strings, and you already explained this above in the docs.

File: algorithms/strings/text_justification.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            for j in range(len(rowWords)):
+                row += rowWords[j]
+                if j != len(rowWords) - 1:
+                    row += ' ' * (1 + spaceNumOfEachInterval)
+                if spaceNumRest > 0:
+                    row += ' '
+                    spaceNumRest -= 1
+        # row with only one word
+        else:
+            row+=rowWords[0]
```
Comment (goswami-rahul) [inline algorithms/strings/text_justification.py]: Leave spaces before and after ` += `.

File: algorithms/strings/text_justification.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        row = ""
+        # if the row is the last
+        if index == len(words):
+            for word in rowWords:
+                row += (word + ' ')
+            row = row[:-1]
+            row += ' ' * (maxWidth - len(row))
+        # not the last row and more than one word
+        elif len(rowWords) != 1:
+            spaceNum = maxWidth - rowLen
```
Comment (goswami-rahul) [inline algorithms/strings/text_justification.py]: Please use `snake_case`, wherever you used `CamelCase`.

File: algorithms/strings/text_justification.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    '''
+    ret=[]  #return value
+    rowLen = 0  # current length of strs in a row
+    rowWords = []  # current words in a row
+    index = 0   # the index of current word in words
+    isFirstWord = True  # is current word the first in a row
+    while index < len(words):
+        while rowLen <= maxWidth and index < len(words):
+            if len(words[index]) > maxWidth:
+                raise Exception("there exists word whose length is larger than maxWidth")
```
Comment (goswami-rahul) [inline algorithms/strings/text_justification.py]: maybe `raise ValueError(...)` would be better option here?

File: algorithms/strings/text_justification.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
+'''
```
Comment (goswami-rahul) [inline algorithms/strings/text_justification.py]: Also, use triple _double_ quotes for documentation. `"""..."""`

Comment (LJXLJXLJX) [reply algorithms/strings/text_justification.py]: thx for your advice !
I have modified them based on your comments.

**Pull request merged**